parser/src/main/scala/hmda/parser/fi/lar/LarCsvParser.scala

```diff
@@ -2,53 +2,59 @@ package hmda.parser.fi.lar
 
 import hmda.model.fi.lar._
 
-import scala.util.Try
+import scala.collection.immutable.ListMap
+import scalaz._
+import scalaz.Scalaz._
+import scala.util.{ Failure, Success, Try }
 
 object LarCsvParser {
   def apply(s: String): Either[List[String], LoanApplicationRegister] = {
-    val larErrors = checkLar(s)
+    val values = s.split('|').map(_.trim)
+    val larErrors = checkLar(values.toList)
+    val parseResults = larErrors.disjunction
```

Comment:
(Forgive me if I already said this; I thought I typed a similar comment earlier today, but I don't see it)

Could we use a `match` statement here to match whether `larErrors` is a Success or Failure? I haven't tried it, but it seems like that could replace the `if`/`else` that tests whether it's a right or left value.